crates/bazed-core/src/buffer.rs

```diff
@@ -1,3 +1,14 @@
+//! Buffer manages all editing operations on the text structure,
+//! including adjusting carets and selections when edits happen and keeping an undo-history.
+//!
+//! # Some terminology
+//! - **character index**: Index of a character in the text.
+//! - **character offset**: (often just "offset") a point in between two characters in the text,
+//!   or the points directly before the first and directly behind the last character.
```

Comment:
I don't understand this explanation.

Reply:
it's less of an explanation and more of a definition :eyes: